smartsim/_core/launcher/step/pbsStep.py

```diff
@@ -49,8 +50,10 @@ def __init__(self, name: str, cwd: str, batch_settings: QsubBatchSettings) -> No
         self.managed = True
 
     @property
-    def batch_settings(self) -> QsubBatchSettings:
-        return self.step_settings
+    def batch_settings(self) -> BatchSettings:
+        if isinstance(self.step_settings, BatchSettings):
+            return self.step_settings
+        raise TypeError("Batch settings must be subtype of BatchSettings")
```

Comment:
Should we add `f"but it is of type {type(self.step_settings}"` to the error message to help the user understand the error? This comment applies to similar `TypeError`s added in this PR.

Reply:
Done